### ISSUE TITLE
Polygon inputs for training custom dataset

### ISSUE THREAD

**Issue by harshall28** [lvpengyuan/masktextspotter.caffe2#52]
Hi,

If I want to train the model on custom dataset, how can I give coordinates of polygon for the text annotations, as in ICDAR2013 and ICDAR2015, the coordinates of only quadrangle are given, i.e coordinates of the 4 corner points. But my question is what if I want to give coordinates of more than 4 points like in the case of curved text. Please let me know how can I do that.

Thanks in advance.

**Comment by harshall28** [comment]
Can you please advice how can I train the model on totaltext_train dataset?

**Comment by eepaul** [comment]
I have the same question too.
in load_gt_from_txt, 
`
if self.name == 'totaltext_train':
  polygons.append(np.array((list(loc[0])*4)[:8]))
    segmentations.append([list(loc[0])])
`

so for totaltext_train, polygons only take the first 4 points(8 numbers). and thus the ploygon would not be something like this(in white) 
![image](https://user-images.githubusercontent.com/1825647/64672324-bd284e00-d49d-11e9-92fa-97e968e5e45e.png)

**Comment by harshall28** [comment]
Exactly, this is what I too had seen, a comment from the authors would be appreciated.

**Comment by PanYuHan** [comment]
can you  run through the model?
